### ISSUE TITLE
Failed to register the window class, resulting in program exit

### ISSUE THREAD

**Issue by 1821746019** [rounk-ctrl/win32-darkmode#2]
I am very eager to enable dark mode for everything, including Bandizip and Process Hacker.
![utools_1710663215697](https://github.com/rounk-ctrl/win32-darkmode/assets/107804044/7f59f855-58b9-412d-bb5d-af3d93eac71a)

When I launch the demo, Controls.exe, no window is created or shown. Upon debugging, I discovered that MyRegisterClass returns 0, indicating that the window class registration failed.

I look forward to your response. I would appreciate it if you could provide me with some guidance on how to achieve my goals.
![image](https://github.com/rounk-ctrl/win32-darkmode/assets/107804044/ffeab91c-38b7-411a-aa2f-e8119db10712)
![image](https://github.com/rounk-ctrl/win32-darkmode/assets/107804044/f4c99342-a42d-431c-965e-7f52a7e5b0c4)
![image](https://github.com/rounk-ctrl/win32-darkmode/assets/107804044/1375e1f7-4a89-49d2-85cb-718f04fc977e)

**Comment by rounk-ctrl** [comment]
weird, it works for me. maybe its failing to get the theme color.

try replacing at line 28, 29, 30 with
```cpp
    HTHEME them = OpenThemeData(0, IsDarkMode ? L"DarkMode::ItemsView" : L"ItemsView");
    GetThemeColor(them, 0, 0, TMT_FILLCOLOR, &clr);
    GetThemeColor(them, 0, 0, TMT_TEXTCOLOR, &clrTxt);
```

**Comment by rounk-ctrl** [comment]
oh, also build in release. for whatever reason idk it fails to work when you build with debug

**Comment by 1821746019** [comment]
There is the same result after replacing in all the build configuarations
![image](https://github.com/rounk-ctrl/win32-darkmode/assets/107804044/c10f310c-1f0b-4bbf-9a71-4a1f11a779fe)

**Comment by 1821746019** [comment]
Is it normal that both szTitle and szWindowClass are empty string after `LoadStringW`?
![image](https://github.com/rounk-ctrl/win32-darkmode/assets/107804044/90974037-f2e1-454d-92cf-2291670bb826)

**Comment by rounk-ctrl** [comment]
weird, does this work?
[Controls.zip](https://github.com/rounk-ctrl/win32-darkmode/files/14811109/Controls.zip)

**Comment by 1821746019** [comment]
Yes, it works. How did you do it?
![Clip_2024-03-31_13-57-22](https://github.com/rounk-ctrl/win32-darkmode/assets/107804044/54071373-d65f-4c9d-827a-07dcdb054446)

**Comment by rounk-ctrl** [comment]
i just built it in x64 release...

**Comment by 1821746019** [comment]
I cloned the repository again and opened it in VS2022. Without modifying any configurations, I built the solution. A window appeared, but the dark mode seemed not to work properly.
![Clip_2024-04-02_09-07-04](https://github.com/rounk-ctrl/win32-darkmode/assets/107804044/2132180e-76fd-4879-9c95-f38b3a8a0d42)

**Comment by TimoKunze** [comment]
> When I launch the demo, Controls.exe, no window is created or shown. Upon debugging, I discovered that MyRegisterClass returns 0, indicating that the window class registration failed.

In `MyRegisterClass` change `WNDCLASSEXW wcex;` to `WNDCLASSEXW wcex = { 0 };`

**Comment by Wallby** [comment]
> > When I launch the demo, Controls.exe, no window is created or shown. Upon debugging, I discovered that MyRegisterClass returns 0, indicating that the window class registration failed.
> 
> In `MyRegisterClass` change `WNDCLASSEXW wcex;` to `WNDCLASSEXW wcex = { 0 };`

Yes, Visual Studio writes random numbers to uninitialized memory when ran in Debug configuration. That is why all the uninitialized fields of wcex contain invalid values.
`{ 0 }` is actually 0 for only the first value and default values for all other values. What would be correct/as intended is: `wcex = {}`
